### ISSUE TITLE
Which qt tool is used? 

### ISSUE THREAD

**Issue by kevinchen1223** [BelledonneCommunications/linphone-desktop#36]
Which qt tool is used? mingw53_32？
There is no gcc_64 in windows qt............

**Comment by Wescoeur** [comment]
If you want to build the project on Windows, you must have a 32bits Qt version. (64bits is not supported at this moment only on Windows.) And mingw.

**Comment by kevinchen1223** [comment]
Thank you very much for answering me！
Yes, I installed qt-opensource-windows-x86-5.9.0.exe and mingw.
And set up Qt5_DIR = C: \ Qt \ Qt5.9.0 \ 5.9 \ mingw53_32 \ lib \ cmake \ Qt5
                  PATH = C: \ Qt \ Qt5.9.0 \ 5.9 \ mingw53_32 \ bin
Is this right?
But VS15 tells me a lot of unresolvable external commands, such as ("__declspec (dllimport) public: class QDebug __thiscall QMessageLogger :: info (void) const" (__imp_info @ QMessageLogger @@ QBE? AVQDebug @@ XZ)).

**Comment by kevinchen1223** [comment]
win10, vs2015, qt-opensource-windows-x86-5.9.0.exe,

**Comment by ghismary** [comment]
I updated the README.md file so that it is clearer. You need to install msvc2015 component when installing Qt 5.9. MingGW is only used to build some external dependencies that we cannot build with Visual Studio (mainly libvpx and ffmpeg), but the application is built with Visual Studio so you need the Qt libs for Visual Studio.

**Comment by kevinchen1223** [comment]
i see,i will to try . thanks!

**Comment by kevinchen1223** [comment]
How to increase h264 video encoding/decoding？
when i "python prepare --all-codecs",it tell me
"OpenH264,H.264 video encoding/decoding support with th openh264 library(require license)"
 "X264,H.264 video encoding support with th x264 library(require license)"

**Comment by kevinchen1223** [comment]
I find the LICENSE in the "C:\Users\Administrator\linphone-desktop\submodules\externals\openh264"
But it does not work。just tell me "require license".
what should i do?

**Comment by Wescoeur** [comment]
Use this to clean your project:

    python prepare.py -c

Enable OpenH264:

    python prepare.py -DENABLE_OPENH264=ON -DENABLE_NON_FREE_CODECS=ON

You can rebuild your project after that.

**Comment by kevinchen1223** [comment]
SO_TIMESTAMP Unknown identifier

**Comment by kevinchen1223** [comment]
rtpsession_inet.c
SO_TIMESTAMP Unknown identifier

**Comment by strangerism** [comment]
Hi I run in with the same issue.

I am trying to build from mac.

I have followed the instructions and installed QT 5.9 but I cannot set env variables as there is no gcc_64 folder in the QT install directory.

ls ~/Qt/5.9/
android_armv7	clang_64	ios

having done everything else and running make I got this output
`-- The C compiler identification is AppleClang 8.1.0.8020042
-- The CXX compiler identification is AppleClang 8.1.0.8020042
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- w
orks
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -
- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test SUGGEST_OVERRIDE
-- Performing Test SUGGEST_OVERRIDE - Failed
CMake Error at CMakeLists.txt:282 (find_package):
  By not providing "FindQt5.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5" with any of
  the following names:

    Qt5Config.cmake
    qt5-config.cmake

  Add the installation prefix of "Qt5" to CMAKE_PREFIX_PATH or set "Qt5_DIR"
  to a directory containing one of the above files.  If "Qt5" provides a
  separate development package or SDK, be sure it has been installed.


-- Configuring incomplete, errors occurred!
See also "/Users/massimo/Linphone/linphone-desktop/WORK/desktop/Build/linphoneqt/CMakeFiles/CMakeOutput.log".
See also "/Users/massimo/Linphone/linphone-desktop/WORK/desktop/Build/linphoneqt/CMakeFiles/CMakeError.log".
make[3]: *** [/Users/massimo/Linphone/linphone-desktop/WORK/desktop/Stamp/EP_linphoneqt/EP_linphoneqt-configure] Error 1
make[2]: *** [CMakeFiles/EP_linphoneqt.dir/all] Error 2
make[1]: *** [all] Error 2
make: *** [desktop-build] Error 2
`

**Comment by strangerism** [comment]
I have fixed this by changing those environment variables to point to clang_64 instead of gcc_64

```
Qt5_DIR="~/Qt/5.9/clang_64/lib/cmake"

export PATH="~/Qt/5.9/clang_64/bin/:$PATH"
```

just a recap, the settings above is based on the following development environment 

- Mac OSX Sierra 

- QT5.9 installed using qt-unified-mac-x64-3.0.0-online.dmg installer

**Comment by ghismary** [comment]
I'm closing this issue as you are now able to build.

**Comment by strangerism** [comment]
maybe would be worth to update README with correct PATHs for mac OSX users?

**Comment by kevinchen1223** [comment]
yes!






------------------ 原始邮件 ------------------
发件人: "Massimo Virgilio";<notifications@github.com>;
发送时间: 2017年7月3日(星期一) 下午5:00
收件人: "BelledonneCommunications/linphone-desktop"<linphone-desktop@noreply.github.com>; 
抄送: "Kevin"<380076693@qq.com>; "Author"<author@noreply.github.com>; 
主题: Re: [BelledonneCommunications/linphone-desktop] Which qt tool isused? (#36)




maybe would be worth to update README with correct PATHs for mac OSX users?

**Comment by ghismary** [comment]
The paths in the README.md file are explicitely told to be examples. You need to adapt them to your configuration.

**Comment by strangerism** [comment]
Not having built/compiled in c/c++ code before in my life, I found the instructions quite misleading for what concern this particular point, especially given the fact that there is special instructions for windows around that environment variable but not for mac so general ones would apply I thought.

I learned by myself my mistake and fixed it, some others might get lost and give up, or come here and annoy people people raising unnecessary issues, just like I did initially.

sharing insight and give my feedback is all I did, isn't this what open source communities do?

**Comment by kevinchen1223** [comment]
You do very well, I am also very grateful to your generous reply. If you do not have a reply I need a long time can not figure out how to compile linphone. Thanks for the open source spirit, I will also be able to reply to the issues.






------------------ 原始邮件 ------------------
发件人: "Massimo Virgilio";<notifications@github.com>;
发送时间: 2017年7月5日(星期三) 凌晨0:20
收件人: "BelledonneCommunications/linphone-desktop"<linphone-desktop@noreply.github.com>; 
抄送: "Kevin"<380076693@qq.com>; "Author"<author@noreply.github.com>; 
主题: Re: [BelledonneCommunications/linphone-desktop] Which qt tool isused? (#36)




Not having built/compiled in c/c++ code before in my life, I found the instructions quite misleading for what concern this particular point, especially given the fact that there is special instructions for windows around that environment variable but not for mac so general ones would apply I thought.
 
I learned by myself my mistake and fixed it, some others might get lost and give up, or come here and annoy people people raising unnecessary issues, just like I did initially.
 
sharing insight and give my feedback is all I did, isn't this what open source communities do?
